client/interfaces.go

```diff
@@ -337,6 +337,11 @@ type InstanceServer interface {
 	RawQuery(method string, path string, data interface{}, queryETag string) (resp *api.Response, ETag string, err error)
 	RawWebsocket(path string) (conn *websocket.Conn, err error)
 	RawOperation(method string, path string, data interface{}, queryETag string) (op Operation, ETag string, err error)
+}
+
+// The AgentServer type represents a LXD agent server.
```

Comment:
I think I'd rather we just use `RawQuery` for now.

The `AgentServer` is possibly a good way to do this but the problem I'm seeing here are:
 - It's not actually really implementing `InstanceServer`, just a bit of a weird subset
 - It's possible for the API between agent and LXD to differ and we wouldn't have a good way to handle that in here
 - As is the case here, we may have structs used for the agent API which aren't public and shouldn't be imported by the client
 - We will get function name conflicts. For example, I very well may want to add a `GetMetrics` to the `InstanceServer` API which accesses `/1.0/metrics` and gets me the text value of the metrics. This wouldn't be possible if we're already using `GetMetrics` with a different signature for the `AgentServer` interface.